pybo/pipeline/pipes.py

```diff
@@ -1,13 +1,56 @@
 # coding: utf-8
+import botok
+
+
+def get_chunks(raw_string):
+    chunker = botok.Chunks(raw_string)
+    chunks = chunker.make_chunks()
+    chunks = chunker.get_readable(chunks)
+    return chunks
+
+
+def shelve_info(chunks):
+    shelved = []
+    clean_chunks = []
+
+    syl_count = 0
+    for i, chunk in enumerate(chunks):
+        marker, text = chunk
+        if marker == "TEXT" or marker == "PUNCT":
+            syl_count += 1
+
+        # 2.a. extract transparent chars
+        # TODO: adapt to also include \t as transparent char
+        if "\n" in text:
+            # remove transparent char
+            text = text.replace("\n", "")
+            index = (syl_count, "\n")
+
+            shelved.append(index)
+            clean_chunks.append((marker, text))
+
+        # 2.b. extract any non-bo chunk
+        elif marker != "TEXT" and marker != "PUNCT":
+            index = (syl_count, text)
+            shelved.append(index)
+
+        else:
+            clean_chunks.append(chunk)
+
+    return clean_chunks, shelved
 
 
 def pybo_prep(in_str):
-    # remove all returns from input
-    in_str = in_str.replace("\n", "")
-    return in_str
+    # 1. get chunks
+    chunks = get_chunks(test)
+
+    # 2. shelve needed info
+    chunks, shelved = shelve_info(chunks)
 
+    # 3. tokenize
+    str_for_botok = "".join([c[1] for c in chunks])
 
-# tokenizing method implemented in cli.py
+    return str_for_botok
```

Comment:
@drupchen Is it correct to put shelving logic in `pybo_pred` ?
If so I am having an issue [here](https://github.com/Esukhia/pybo/issues/3#issuecomment-657032363)